functional-tests/src/test/java/org/infinispan/online/service/utils/ReadinessCheck.java

```diff
@@ -4,31 +4,37 @@
 import io.fabric8.openshift.client.OpenShiftClient;
 
 import java.util.Set;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.locks.LockSupport;
 import java.util.stream.Collectors;
 
 public class ReadinessCheck {
 
-   public void waitUntilAllPodsAreReady(OpenShiftClient client, long timeout, TimeUnit unit) {
-      long endTime = TimeUnit.MILLISECONDS.convert(timeout, unit) + System.currentTimeMillis();
-      long backoffCounter = 0;
-      while (System.currentTimeMillis() - endTime < 0) {
-         Set<Pod> notReadyPods = getNotReadyPods(client);
-         if (notReadyPods.isEmpty()) {
-            break;
-         }
-         System.out.println("Some pods are not ready: " + notReadyPods);
-         LockSupport.parkNanos(++backoffCounter * 10_000_000);
-      }
+   private Waiter waiter = new Waiter();
+
+   public void waitUntilTargetNumberOfReplicasAreReady(String podPartialName, int replicas, OpenShiftClient client) {
+      waiter.waitFor(() -> getPodsWithName(podPartialName, client).size() == replicas);
+      System.out.println("SA 2");
```

Comment:
I will remove those after CI run... no worries.. Those are for debugging since I need to know how it behaves in slower environment.